suse_caasp

```diff
@@ -378,20 +382,218 @@ if check_rpm kubernetes-client && [ -f /etc/kubernetes/admin.conf ]; then
     log_cmd $OF 'kubectl version'
     log_cmd $OF 'kubectl api-versions'
     log_cmd $OF 'kubectl config view'
+    log_cmd $OF 'kubectl -n kube-system get pods --field-selector=status.phase=Running'
+    log_cmd $OF 'kubectl get events --sort-by=.metadata.creationTimestamp'
     log_cmd $OF 'kubectl get nodes'
+    log_cmd $OF 'kubectl get all -A'
+    log_cmd $OF 'kubectl get nodes -o yaml'
     if kubectl cluster-info &>/dev/null; then
         timeout 3 kubectl cluster-info dump --output-directory=$KUBECTL_LOG; echo
         find_and_log_files $OF $KUBECTL_LOG
         rm ${KUBECTL_LOG} -fr
     fi
     find_and_conf_files $OF /etc/kubernetes
+    set +a
+fi
+
+#############################################################
+section_header "Checking kubelet service ..."
+OF=kubelet.txt
+plugin_message "All data stored in $OF"
 
+# Fetch kubelet logs
+if check_rpm kubernetes-kubelet; then
     rpm_verify $OF kubernetes-kubelet
-    for K8S_SERVICE in kube-proxy kubelet; do
-        log_cmd $OF "systemctl status --full $K8S_SERVICE"
-        log_cmd $OF "journalctl -u $K8S_SERVICE"
-    done
-    set +a
+    log_cmd $OF "systemctl status --full kubelet"
+    log_cmd $OF "journalctl -u kubelet"
+    log_cmd $OF "cat /var/lib/kubelet/config.yaml"
+    # Should be applicable only to masters
+    find_and_log_files $OF "/etc/kubernetes/manifests"
+fi
+plugin_message Done
+
+#############################################################
+section_header "Checking kube-proxy container service ..."
+OF=kube-proxy.txt
+plugin_message "All data stored in $OF"
+
+# Fetch kube-proxy logs
+if crictl pods | grep kube-proxy &> /dev/null; then
+    # Fetch the kube-proxy container ID
+    kubeproxycontainer=$(crictl ps --label io.kubernetes.container.name=kube-proxy --quiet)
+    # kube-proxy container
+    log_cmd $OF "crictl logs -t $kubeproxycontainer"
+    log_cmd $OF "crictl inspect $kubeproxycontainer"
+    log_cmd $OF "crictl stats --id $kubeproxycontainer"
+    # kube-proxy pod
+    kubeproxypod=$(crictl ps | grep kube-proxy | awk -F ' ' '{ print $9 }')
+    log_cmd $OF "crictl inspectp $kubeproxypod"
+fi
+plugin_message Done
+
```

Comment:
This can be run inside `for` loop for `kube-proxy, cilium-agent, cilium-operator, kured, oidc-gangway, coredns, kube-apiserver, kube-scheduler`

Reply:
Good idea!